alacritty/src/renderer/platform.rs

```diff
@@ -81,21 +81,31 @@ pub fn create_gl_context(
     gl_config: &Config,
     raw_window_handle: Option<RawWindowHandle>,
 ) -> GlutinResult<NotCurrentContext> {
-    let context_attributes = ContextAttributesBuilder::new()
-        .with_context_api(ContextApi::OpenGl(Some(Version::new(3, 3))))
-        .build(raw_window_handle);
-
-    unsafe {
-        if let Ok(gl_context) = gl_display.create_context(gl_config, &context_attributes) {
-            Ok(gl_context)
-        } else {
-            let context_attributes = ContextAttributesBuilder::new()
-                .with_profile(GlProfile::Compatibility)
-                .with_context_api(ContextApi::OpenGl(Some(Version::new(2, 1))))
-                .build(raw_window_handle);
-            gl_display.create_context(gl_config, &context_attributes)
-        }
+    let mut profiles = [
+        ContextAttributesBuilder::new()
+            .with_context_api(ContextApi::OpenGl(Some(Version::new(3, 3))))
+            .build(raw_window_handle),
+        // Try gles3 as well, since it's similar to OpenGL.
+        ContextAttributesBuilder::new()
+            .with_context_api(ContextApi::Gles(Some(Version::new(3, 0))))
+            .build(raw_window_handle),
+        ContextAttributesBuilder::new()
+            .with_profile(GlProfile::Compatibility)
+            .with_context_api(ContextApi::OpenGl(Some(Version::new(2, 1))))
+            .build(raw_window_handle),
+    ]
+    .into_iter();
+
+    // Try the optimal config first.
+    let mut picked_context =
+        unsafe { gl_display.create_context(gl_config, &profiles.next().unwrap()) };
+
+    // Try the fallback ones.
+    while let (Err(_), Some(profile)) = (picked_context.as_ref(), profiles.next()) {
+        picked_context = unsafe { gl_display.create_context(gl_config, &profile) };
     }
```

Comment:
Wouldn't this choose the lowest supported config rather than the highest one?

Reply:
No, because I iterate while `Err`, not while `Ok`? So it stops once something is picked.

Reply:
It stops **before** something is `Ok` though, right?

Reply:
You have initial iteration out side of the loop though?

Reply:
Yeah nvm I wasn't really looking at the code. You're just checking if the current context had any errors or not and reassigning until it doesn't.

I feel like this might be cleaner with iterators though, like `find` or `find_map`?

Reply:
I though about it, but how do you propagate the error out of it?

Reply:
I guess it should be fine.